### PR TITLE
Migrated v0 endpoints for machine and machine_type into v1/obj endpoint

### PR DESCRIPTION
**Pull request by ankitintg**
https://sightmachine.atlassian.net/browse/ENG-3029
Update get_machine and get_machine_type API's of SDK to use v1 endpoints (v1/obj/....)

### CODE REVIEW

**Comment by mrnthv** [comment]
Unit tests are fixed. @mklein0 do review and provide your comments so that we can proceed with merging.

File: smsdk/client.py [hunk 1]
```diff
@@ -48,7 +48,9 @@ def dict_to_df(data, normalize=True):
                 # machine type stats are list
                 cols = [*data[0]]
                 cols.remove("stats")
-                df = json_normalize(data, "stats", cols, record_prefix="stats.")
+                df = json_normalize(
+                    data, "stats", cols, record_prefix="stats.", errors="ignore"
```
Comment (mgoldmansight) [inline smsdk/client.py:52]: Why are we ignoring errors here?

File: tests/machine/test_machine.py [hunk 2]
```diff
@@ -126,3 +126,8 @@ def test_get_machine_schema_types_return_mtype(mocked_types, mocked_machines):
         "stat__test_float",
         "stat__test_string",
     ]
+
+
+def test_get_machines_v1(get_client):
```
Comment (mgoldmansight) [inline tests/machine/test_machine.py:131]: Add a note that is tested against demo and if that env is changed this test has to as well

File: tests/machine_type/test_machine_type.py [hunk 3]
```diff
@@ -91,3 +91,8 @@ def test_get_fields_of_machine_type_types(mocked_machines):
 
     # Verify
     assert names == ["stat__test_float"]
+
+
+def test_get_machines_types_v1(get_client):
```
Comment (mgoldmansight) [inline tests/machine_type/test_machine_type.py:96]: Same comment as with test_get_machines_v1

File: smsdk/client.py (hunk 1 above)
Comment (ankitintg) [reply smsdk/client.py:52]: While getting get_machine_types() with v1 endpoint we are getting `part_types` in response but not always that's why we are getting the key error like this- `KeyError: "Try running with errors='ignore' as key 'part_types' is not always present"
`
Do you think we can do something else to address this issue? like avoid `part_types` from the response or just ignore the errors from `json_normalize` function

File: tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+from smsdk import client
+from requests.sessions import Session
+import pytest
+
+
+# Define all the constants used in the test
+# These values may change for each run.
+API_KEY = "5a73aa5a-1962-4df9-b56e-4a59462f0f00"
+API_SECRETE = "sma_FajgH3VbPu68gwy0PzccvhyGRyy1a8CCHhhvy6ooeg1O_"
```
[the remark this answers is not in the document]
Comment (ankitintg) [reply tests/conftest.py:9]: Do we want to set the variable and access it from os.environ() or should we just create const.py or JSON file and access these variables from there?
Need your comment on this.

File: smsdk/config/api_endpoints.json [hunk 5]
```diff
@@ -12,10 +12,12 @@
   },
   "MachineType": {
     "url" : "/api/machinetype",
+    "url_v1" : "/v1/obj/machine_type",
     "fields": "/v1/selector/datatab/cycle/{}/field"
   },
   "Machine": {
-    "url" : "/api/machine"
+    "url" : "/api/machine",
```
[the remark this answers is not in the document]
Comment (ankitintg) [reply smsdk/config/api_endpoints.json]: No need of old URLs, hence removed in latest changes.

File: tests/machine_type/test_machine_type.py (hunk 3 above)
Comment (ankitintg) [reply tests/machine_type/test_machine_type.py:101]: Done

File: tests/machine/test_machine.py (hunk 2 above)
Comment (ankitintg) [reply tests/machine/test_machine.py:136]: Done

File: smsdk/config/api_endpoints.json [hunk 6]
```diff
@@ -4,26 +4,23 @@
   },
   "Cycle": {
     "url_v1": "/v1/datatab/cycle",
-    "url": "/api/cycle",
     "alt_url": "/api/cycle"
```
Comment (mklein0) [inline smsdk/config/api_endpoints.json:7]: Should this be removed?

File: smsdk/config/api_endpoints.json [hunk 7]
```diff
@@ -4,26 +4,23 @@
...
   },
   "Factory": {
     "url" : "/api/factory"
```
Comment (mklein0) [inline smsdk/config/api_endpoints.json:10]: This probably needs to be a separate ticket.  Maybe related to https://sightmachine.atlassian.net/browse/ENG-3228

**Review by mklein0 — APPROVED** [review]

**Pull request merged**